_posts/2025-09-15-quarkus-a2a-java-0-3-0-beta-release.adoc

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: 'A2A Java SDK: Support for the REST Transport is Now Here'
+date: 2025-09-15
+tags: ai a2a
+synopsis: 'Today, we released A2A Java SDK 0.3.0.Beta1 which adds support for the HTTP+JSON/REST transport.'
+author: fjuma
+---
+
+Today, we've released A2A Java SDK 0.3.0.Beta1 which introduces support for the HTTP+JSON/REST transport.
+
+Our last https://quarkus.io/blog/quarkus-a2a-java-0-3-0-alpha-release/[blog post] covered what's new in the 0.3.0 version of the A2A Java SDK. In this post, we'll focus on how to make use of the new HTTP+JSON/REST transport for both A2A server agents and clients.
+
+== Configuring an A2A Server Agent to Support the REST Transport
+
+To enable your A2A server agent to support communication using HTTP+JSON/REST, add the following dependency:
+
+[NOTE]
+====
+The `io.github.a2asdk` `groupId` is temporary and will likely change for future releases. Keep an eye on the `a2a-java` https://github.com/a2aproject/a2a-java/blob/main/README.md[README] for up-to-date documentation.
+====
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.github.a2asdk</groupId>
+    <artifactId>a2a-java-sdk-reference-rest</artifactId> <1>
+</dependency>
+----
+
+<1> `a2a-java-sdk-reference-rest` provides access to the core classes that make up the A2A specification and provides the HTTP endpoints that implement the A2A protocol using the HTTP+JSON/REST transport.
+
+After adding this dependency, simply update your agent card to declare support for this new transport:
+
+[source,java]
+----
+@Produces
+@PublicAgentCard
+public AgentCard agentCard() {
+    return new AgentCard.Builder()
+             .url(YOUR_HTTP_JSON_URL) <1>
+             .preferredTransport(TransportProtocol.HTTP_JSON.asString()) <2>
+             .additionalInterfaces(List.of(
+                     new AgentInterface(TransportProtocol.HTTP_JSON.asString(),
+                                        YOUR_HTTP_JSON_URL) <3>
```

Comment:
Is that what is used in the case of negotiation? If so, why do I have to repeat it? (The order could be used, no?)